Framework/Kernel/src/ConfigService.cpp

```diff
@@ -1582,6 +1582,44 @@ void ConfigServiceImpl::setDataSearchDirs(const std::string &searchDirs) {
 }
 
 /**
+ *  Appends the passed subdirectory path to the end of each of data
+ *  search dirs and adds these new dirs to data search directories
+ *  @param subdir :: the subdirectory path to add (relative)
+ */
+void ConfigServiceImpl::appendDataSearchSubDir(const std::string &subdir) {
+  if (subdir.empty())
+    return;
+
+  Poco::Path subDirPath;
+  try {
+    subDirPath = Poco::Path(subdir);
+  } catch (Poco::PathSyntaxException &) {
+    return;
+  }
+
+  if (!subDirPath.isDirectory() || !subDirPath.isRelative()) {
+    return;
+  }
+
+  std::vector<std::string> newDataDirs = m_DataSearchDirs;
```

Comment:
This is the right thing to do but I think we can tidy it up slightly with some C++11 syntax:
```c++
auto newDataDirs = m_DataSearchDirs;
for (const auto & path : m_DataSearchDirs) {
  Poco::Path newDirPath;
  try {
    newDirPath = Poco::Path(path);
    newDirPath.append(subDirPath);
    newDataDirs.push_back(newDirPath.toString());
  } catch (Poco::PathSyntaxException &) {
    continue;
  }
}
```

It saves having to worry about the loop termination and types of iterations etc.